[CI] CoordinationDiagnosticsServiceIT testBlockClusterStateProcessingOnOneNode failing

Originally found in commit b8ba76f of PR 88639, but also found it reproduces on current master (commit b724039).

Error:
```
  2> REPRODUCE WITH: ./gradlew ':server:internalClusterTest' --tests "org.elasticsearch.cluster.coordination.CoordinationDiagnosticsServiceIT.testBlockClusterStateProcessingOnOneNode" -Dtests.seed=7F4CA1286BC46227 -Dtests.locale=en-PH -Dtests.timezone=US/Indiana-Starke -Druntime.java=18
  2> java.lang.AssertionError:
    Expected: is <2>
         but: was <0>
        at __randomizedtesting.SeedInfo.seed([7F4CA1286BC46227:ADB3E4A652F0D4EB]:0)
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
        at org.junit.Assert.assertThat(Assert.java:956)
        at org.junit.Assert.assertThat(Assert.java:923)
        at org.elasticsearch.cluster.coordination.CoordinationDiagnosticsServiceIT.lambda$testBlockClusterStateProcessingOnOneNode$3(CoordinationDiagnosticsServiceIT.java:82)
        at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:1104)
        at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:1077)
        at org.elasticsearch.cluster.coordination.CoordinationDiagnosticsServiceIT.testBlockClusterStateProcessingOnOneNode(CoordinationDiagnosticsServiceIT.java:80)
  2> NOTE: leaving temporary files on disk at: /home/kingherc/repos/elasticsearch/server/build/testrun/internalClusterTest/temp/org.elasticsearch.cluster.coordination.CoordinationDiagnosticsServiceIT_7F4CA1286BC46227-001
  2> NOTE: test params are: codec=Asserting(Lucene92): {type=Lucene90}, docValues:{}, maxPointsInLeafNode=522, maxMBSortInHeap=7.595551898127974, sim=Asserting(RandomSimilarity(queryNorm=true): {}), locale=en-PH, timezone=US/Indiana-Starke 
  2> NOTE: Linux 5.15.0-41-generic amd64/Oracle Corporation 18.0.2 (64-bit)/cpus=8,threads=1,free=403315816,total=536870912
  2> NOTE: All tests run in this JVM: [CoordinationDiagnosticsServiceIT]
```

**Build scan:**
https://gradle-enterprise.elastic.co/s/73jdeafslzxgy/tests/:server:internalClusterTest/org.elasticsearch.cluster.coordination.CoordinationDiagnosticsServiceIT/testBlockClusterStateProcessingOnOneNode

**Reproduction line:**
`./gradlew ':server:internalClusterTest' --tests "org.elasticsearch.cluster.coordination.CoordinationDiagnosticsServiceIT.testBlockClusterStateProcessingOnOneNode" -Dtests.seed=7F4CA1286BC46227 -Dtests.locale=en-PH -Dtests.timezone=US/Indiana-Starke -Druntime.java=17`

**Applicable branches:**
master

**Reproduces locally?:**
Yes

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.cluster.coordination.CoordinationDiagnosticsServiceIT&tests.test=testBlockClusterStateProcessingOnOneNode

**Failure excerpt:**
```
java.lang.AssertionError: 
Expected: is <2>
     but: was <1>

  at __randomizedtesting.SeedInfo.seed([7F4CA1286BC46227:ADB3E4A652F0D4EB]:0)
  at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
  at org.junit.Assert.assertThat(Assert.java:956)
  at org.junit.Assert.assertThat(Assert.java:923)
  at org.elasticsearch.cluster.coordination.CoordinationDiagnosticsServiceIT.lambda$testBlockClusterStateProcessingOnOneNode$3(CoordinationDiagnosticsServiceIT.java:81)
  at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:1104)
  at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:1077)
  at org.elasticsearch.cluster.coordination.CoordinationDiagnosticsServiceIT.testBlockClusterStateProcessingOnOneNode(CoordinationDiagnosticsServiceIT.java:80)
  at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(NativeMethodAccessorImpl.java:-2)
  at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
  at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  at java.lang.reflect.Method.invoke(Method.java:568)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:44)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:843)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:490)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:850)
  at java.lang.Thread.run(Thread.java:833)

```

--- a/server/src/internalClusterTest/java/org/elasticsearch/cluster/coordination/CoordinationDiagnosticsServiceIT.java
+++ b/server/src/internalClusterTest/java/org/elasticsearch/cluster/coordination/CoordinationDiagnosticsServiceIT.java
@@ -35,7 +35,6 @@ public class CoordinationDiagnosticsServiceIT extends ESIntegTestCase {
         internalCluster().setBootstrapMasterNodeIndex(0);
     }
 
-    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/89015")
     public void testBlockClusterStateProcessingOnOneNode() throws Exception {
         /*
          * This test picks a node that is not elected master, and then blocks cluster state processing on it. The reason is so that we
@@ -48,6 +47,7 @@ public class CoordinationDiagnosticsServiceIT extends ESIntegTestCase {
         assertThat(nodeNames, hasItem(master));
         String blockedNode = nodeNames.stream().filter(n -> n.equals(master) == false).findAny().get();
         assertNotNull(blockedNode);
+        ensureStableCluster(3);
 
         DiscoveryNodes discoveryNodes = internalCluster().getInstance(ClusterService.class, master).state().nodes();
         Set<DiscoveryNode> nodesWithoutBlockedNode = discoveryNodes.getNodes()

